### DISCUSSION TITLE
Is there any a more efficient way Copying data from C# array to Python list or Numpy array?

### DISCUSSION THREAD

**Discussion by bruceYangxianglin** [pythonnet/pythonnet#1912] [Q&A]
### Environment

-   Pythonnet version: pythonnet 2.5.2
-   Python version: python 3.8
-   Operating System: win10
-   .NET Runtime: .net 4.0

### Details

-   Copying data from C# array to Python list or Numpy array take much time.  Is there any a more efficient way do that?

-   arInput = list(CShapeClass.IntArrayDataFromCShape)

```python
    print('Extract data From System.Byte[] with time: {} ms'.format(round(tSpend * 1000, 3)))
```

-   The System.Byte[] has 12288000 Bytes Members. And it takes about 4 seconds to extract data to python list.

![Extract data time](https://user-images.githubusercontent.com/93565254/184545152-48ab9815-bf0f-441e-8436-a91ad9193d1f.png)

**Comment by RobVanDerHoeven** [comment]
I use the code from Robert McLeod:

https://github.com/pythonnet/pythonnet/issues/514#issuecomment-350375105
